src/bun.js/node/node_fs.zig

```diff
@@ -141,6 +143,185 @@ pub const Async = struct {
         }
     };
 
+    fn NewUVFSRequest(comptime ReturnType: type, comptime ArgumentType: type, comptime FunctionEnum: NodeFSFunctionEnum) type {
+        if (!Environment.isWindows) {
+            return NewAsyncFSTask(ReturnType, ArgumentType, @field(NodeFS, @tagName(FunctionEnum)));
+        }
+        switch (FunctionEnum) {
+            .open,
+            .close,
+            .read,
+            .write,
+            .readv,
+            .writev,
+            => {},
+            else => return NewAsyncFSTask(ReturnType, ArgumentType, @field(NodeFS, @tagName(FunctionEnum))),
+        }
+
+        comptime std.debug.assert(Environment.isWindows);
+        return struct {
+            promise: JSC.JSPromise.Strong,
+            args: ArgumentType,
+            globalObject: *JSC.JSGlobalObject,
+            req: uv.fs_t = std.mem.zeroes(uv.fs_t),
+            result: JSC.Maybe(ReturnType),
+            ref: bun.Async.KeepAlive = .{},
+            tracker: JSC.AsyncTaskTracker,
+
+            pub const Task = @This();
+
+            pub const heap_label = "Async" ++ bun.meta.typeBaseName(@typeName(ArgumentType)) ++ "UvTask";
+
+            pub fn create(globalObject: *JSC.JSGlobalObject, args: ArgumentType, vm: *JSC.VirtualMachine) JSC.JSValue {
+                var task = bun.new(Task, Task{
+                    .promise = JSC.JSPromise.Strong.init(globalObject),
+                    .args = args,
+                    .result = undefined,
+                    .globalObject = globalObject,
+                    .tracker = JSC.AsyncTaskTracker.init(vm),
+                });
+                task.ref.ref(vm);
+                task.args.toThreadSafe();
+
+                task.tracker.didSchedule(globalObject);
+
+                const log = bun.sys.syslog;
+                const loop = uv.Loop.get();
+                task.req.data = task;
+                switch (FunctionEnum) {
+                    .open => {
+                        var buf: bun.PathBuffer = undefined; // uv copies path so its okay to use stack memory for this
+                        const path = args.path.sliceZ(&buf);
+
+                        var flags: c_int = @intFromEnum(args.flags);
+                        flags = uv.O.fromBunO(flags);
+
+                        var mode: c_int = args.mode;
+                        if (mode == 0) mode = 0o644;
+
+                        bun.debugAssert(uv.uv_fs_open(loop, &task.req, path.ptr, flags, mode, &uv_callback) == .zero);
```

Comment:
Don't use `debugAssert` for side-effectful things. Let's copy into a local variable. I'm also not sure if these should be debugAssert, as users may provide invalid input and that's not an assertion error.

```
const rc = uv.uv_fs_open(loop, &task.req, path.ptr, flags, mode, &uv_callback);
bun.debugAssert(rc == .zero);